pkg/fsutils/fsutils.h

```diff
@@ -0,0 +1,29 @@
+#ifndef _FSUTILS_H
+#define _FSUTILS_H
+
+#include <stdlib.h>
+#include <sys/types.h>
+#include <dirent.h>
+
+static int supports_d_type(const char* path) {
+  int result = 1; // true
+  DIR *d = opendir(path);
+  if (d == NULL) {
+    result = -1;
+    goto ret;
+  }
+  struct dirent *ent;
+  while ((ent = readdir(d)) != NULL) {
+    if (ent->d_type == DT_UNKNOWN) {
```

Comment:
It should be possible to do this without c. https://golang.org/pkg/syscall/#Dirent is exposed in go.